### DISCUSSION TITLE
How many iteration should I determine in param.json?

### DISCUSSION THREAD

**Discussion by zhao-w-en** [deepmodeling/dpgen#652] [Q&A]
Hello,developers，
    While I was setting up the dpgen input file, there were some settings that looked like experience items.

-  I wonder how to determine the **number of iterations** of dpgen.  Ideally it should stop until dpgen can't pick a new configuration from extrapolation procedure, but how can I determine how many iterations should I set before I run the dpgen?

- In addition to this, I was curious about the `"stop_batch"` in training part, and it looks like the item in the example file was set to the order of 10000. That's a lot less than  `"stop_batch"` in the **DP train.**

- I noticed that some papers had very little initial training data, usually a few hundred to a few thousand configurations.  

> Free energy of proton transfer at the water–TiO2 interface from ab initio deep potential molecular dynamics. DOI: [10.1039/C9SC05116C](https://doi.org/10.1039/C9SC05116C) (Edge Article) [Chem. Sci.](https://doi.org/10.1039/2041-6539/2010), 2020, 11, 2335-2341

Does this mean that if I spend enough time doing dpgen, I can end up with a DP potential with ideal precision?

Yours,
Zhaowen

**Comment by njzjz** [comment]
If you have never tried similar systems, I don't think you can determine the number of iterations and other hyper-parameters **before** starting the workflow of the DP-GEN.

**Reply by zhao-w-en** [reply]
I have tried several iterations, but it seems that the "failed" ratio does not decrease all the time and the "Accurate" configuration does not appear. I used **DP train** to a good potential  for the same system but have no idea on  how to tweak dpgen settings. It is worth noting that each iteration also takes much time.

**Reply by njzjz** [reply]
You should firstly determine a proper range for accurate frames and candidates. You may check the same indicators on the training set. Of course, you should make sure that the model has been trained into an accepted level.

**Comment by AnguseZhang** [comment]
For training steps in DP-GEN protocols, we would usually suggest performing relatively short trainings during iterations, such as 400000 to 1000000 steps. When all iterations in DP-GEN finish, we suggest perform a long training, such as 8000000 steps, to achieve better training accuracy.
